taxcalc/parameters.py

```diff
@@ -1,170 +1,496 @@
-import numpy as np
+"""
+OSPC Tax-Calculator federal tax policy Parameters class.
+"""
+# CODING-STYLE CHECKS:
+# pep8 --ignore=E402 parameters.py
+# pylint --disable=locally-disabled parameters.py
+
 from .utils import expand_array
 import os
 import json
-from pkg_resources import resource_stream, Requirement
-
-DEFAULT_START_YEAR = 2013
 
 
 class Parameters(object):
-
-    CUR_PATH = os.path.abspath(os.path.dirname(__file__))
-    PARAM_FILENAME = "params.json"
-    params_path = os.path.join(CUR_PATH, PARAM_FILENAME)
-
-    # Mapping of year to inflation rate
+    """
+    Constructor for the federal tax policy parameters class.
+
+    Parameters
+    ----------
+    parameter_dict: dictionary of PARAM:DESCRIPTION pairs
+        dictionary of policy parameters; if None, default policy
+        parameters are read from the params.json file.
+
+    start_year: integer
+        first calendar year for historical policy parameters.
+
+    num_years: integer
+        number of calendar years for which to specify policy parameter
+        values beginning with start_year.
+
+    inflation_rates: dictionary of YEAR:RATE pairs
+        variable inflation rates used to project future policy parameter
+        values; if None, default inflation rates (specified below) are used.
+
+    Raises
+    ------
+    ValueError:
+        if num_years is less than one.
+        if len(inflation_rates) is not equal to num_years.
+        if min(inflation_rates.keys()) is not equal to start_year.
+
+    Returns
+    -------
+    class instance: Parameters
+    """
+
+    PARAMS_FILENAME = 'params.json'
```

Comment:
The tradeoff that I see is that if we keep these constants at module level scope we don't have to type `Parameters.PARAMS_FILENAME` inside this module, so the code is just more concise. Also, the name of the file in which we store the data seems to me to be more of a characteristic of our package itself, and not particular to the Parameters class. The filename is about the physical layout of our package data; the class is an abstraction. Typically, variables like this are scoped inside classes to prevent pollution of a global namespace. This is more of an issue when you have languages with a true preprocessor, like C++ or Java. In Python, the only way that these variables would "leak" is if you did this in some other module:

```
from parameters import *
```

which is an atypical thing to do anyway. Scoping inside the class would keep these variables "inside" in that case, but that doesn't seem like a compelling argument to me. So I was wondering if there was another reason.